datafusion/common/src/scalar/mod.rs

```diff
@@ -3415,6 +3434,58 @@ impl ScalarValue {
                 .map(|sv| sv.size() - size_of_val(sv))
                 .sum::<usize>()
     }
+
+    /// Performs a deep clone of the ScalarValue, creating new copies of all nested data structures.
+    /// This is different from the standard `clone()` which may share data through `Arc`.
+    /// Aggregation functions like `max` will cost a lot of memory if the data is not cloned.
+    pub fn deep_clone(&self) -> Self {
```

Comment:
I think `deep_clone` is a fine name. Another one might be `force_clone` or something to indicate it is cloning the underlying data (not just array refs)